prime-router/src/main/kotlin/cli/ConvertFileCommands.kt

```diff
@@ -0,0 +1,106 @@
+package gov.cdc.prime.router.cli
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.output.TermUi
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.path
+import de.m3y.kformat.Table
+import de.m3y.kformat.table
+import gov.cdc.prime.router.FileSource
+import gov.cdc.prime.router.Metadata
+import gov.cdc.prime.router.serializers.CsvSerializer
+import kotlin.io.path.inputStream
+import kotlin.io.path.nameWithoutExtension
+
+/**
+ * Converts a file into a different output format
+ */
+class ConvertFileCommands(
+    private val metadataInstance: Metadata? = null
+) : CliktCommand(
+    name = "convert-file",
+    help = """
+        
+    """
+) {
+    /**
+     *
+     */
+    private val tableHeaders by option(
+        "--headers",
+        metavar = "",
+        help = ""
+    )
+
+    /**
+     *
+     */
+    private val fieldNames by option(
+        "--fields"
+    ).required()
+
+    /**
+     *
+     */
+    private val inputSchema by option(
+        "--input-schema"
+    ).required()
+
+    /**
+     *
+     */
+    private val outputFile by option(
+        "--output-file"
+    ).path(mustExist = false, canBeDir = false)
+
+    /**
+     *
+     */
+    private val inputFile by option(
+        "--input-file"
+    ).path(mustExist = true, canBeDir = false).required()
+
+    /**
+     *
+     */
+    override fun run() {
+        // get a metadata instance
+        val metadata = metadataInstance ?: Metadata.getInstance()
+        val schemaName = inputSchema.lowercase()
+        val schema = metadata.findSchema(schemaName) ?: error("Schema $schemaName is not found")
+        val csvSerializer = CsvSerializer(metadata)
+        val report = csvSerializer.readExternal(
+            schema.name,
+            inputFile.inputStream(),
+            FileSource(inputFile.nameWithoutExtension)
+        ).report
+        val fields = fieldNames.split(",")
+        val rows = mutableListOf<Array<String>>()
+        for (i in 0 until report?.itemCount!!) {
+            fields.map {
+                report.getString(i, it) ?: ""
+            }.toTypedArray().let { rows.add(it) }
+        }
+        // our headers
+        val headers = (tableHeaders ?: fieldNames).split(",").toTypedArray()
+        // a default value row
+        val columnCount = headers.count()
+        val defaultValues = Array(columnCount) { "" }
+        if (rows.isEmpty()) {
+            rows.add(defaultValues)
+        }
+        // render a table
+        val table = table {
+            hints {
+                borderStyle = Table.BorderStyle.SINGLE_LINE
+            }
+            header(*headers)
+            rows.map {
+                row(*it)
+            }
+        }.render()
+        // output it
+        TermUi.echo(table)
+    }
+}
```

Comment:
I actually forgot I put this into the application until I did this review, but what this lets you do is ingest a file, for example a STRAC CSV file, and then tell it to output some values based on the schema field names and it will output them as a table. This isn't the end of this feature, but I used it quick to extract out values for PA's review spreadsheets they want whenever you send in files. 

I intend to extend this also read HL7 files and let you pass in the segments you want for a row, as well as specify that you want it to kick out a CSV file from the values you've selected. Good things to come.